### DISCUSSION TITLE
Docking Port Circuit Boxes are incredibly vulnerable to griefing by deleting their contents on all Submarines

### DISCUSSION THREAD

**Discussion by WJohn** [FakeFishGames/Barotrauma#14151] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

All docking ports on all submarines feature a circuit box that controls the docking mechanism. These circuit boxes are out in the open beneath the docking ports, and anyone is able to walk up to them and delete their contents in 1 second flat. While the circuit boxes cannot be wrenched off the walls, the contents have no such safeguards. The following video illustrates the issue:

https://github.com/FakeFishGames/Barotrauma/assets/5988156/1082e938-4ea4-4973-98b9-2d3865325326

The following questions are asked:
**1.** Why would the crew need to be able to mess with the ability to dock and end the round?
**2.** If they are meant to mess with it, why is the docking circuit needlessly over-engineered and complicated?
**3.** If they are not meant to mess with it as the locked wrenching implies, why is it exposed at all?

Hiding the Circuit Box alone will not fix the issue outright, as all it does is delay someone through to welding the spot in the wall where it can be accessed and deleted again. Locking both the wrenching and wire editing of the Circuit Box prevents players from messing with it in interesting ways. Finally, if they want to mess with it, it needs to be extremely simple to remember how to fix it, and be as simple of a circuit as possible.

Finally, the Docking Circuit Box's over-engineering fails at its purpose. It is designed to prevent a bit of water from flooding in the submarine when undocking, but this assumes that the submarine is the one that undocks from the ship or respawn shuttle in question. In the vast majority of cases, it is the docking shuttle that automatically undocks from the submarine when all players have left it. This completely bypasses the Docking Port having any time to close, and so some water will get in anyway. The complicated circuitry achieves nothing.

Given that 1.4 has reduced flood rate by roughly 50%, this is even less of an issue than ever before. Circuit boxes could use a much simpler circuit, as shown here:
![simplified_docking_circuit](https://github.com/FakeFishGames/Barotrauma/assets/5988156/b71a3869-a28b-4d9f-9cad-3ae2e684f3fa)

This docking circuit mirrors the main benefits of the new circuit, in that a button can be easily installed to toggle the hatch's state open or closed, or any other method of door opening while undocked, without affecting the ability to properly dock and keep the hatch open. Additionally, this only uses 2 Components, down from the 9 the previous circuit used.

I have attached the assembly below, as well as instructions on how to hook it up:
<details><summary><b>Simplified Docking Circuit Box Item Assembly: (click and drag the contents into the Submarine Editor)</b></summary>

Here is the Simplified Circuit Box item assembly download, just drag and drop the "Simplified Docking Circuit.xml" into the Submarine editor and follow the picture instructions below:
[Simplified Docking Circuit.zip](https://github.com/user-attachments/files/15877333/Simplified.Docking.Circuit.zip)

<b>It is crucial that the Circuit Box is moved inside the wall where players cannot so easily mess with it!</b>

<img src="https://github.com/FakeFishGames/Barotrauma/assets/5988156/ad16d94b-3c76-488c-bbb1-4739378fe139">
<img src="https://github.com/FakeFishGames/Barotrauma/assets/5988156/bf914691-5a7e-47a4-8c56-64e68ac5b672">
<img src="https://github.com/FakeFishGames/Barotrauma/assets/5988156/437b755a-cee8-479a-833f-3c1d988f9c83">

</details>

### Reproduction steps

1. Start a new game in singleplayer or multiplayer
2. Walk up to the Docking Port's Circuit Box
3. Open the Circuit Box with a free hand
4. Select all the Components in the Circuit Box
5. Press the DEL key

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Happens in both single player and multiplayer

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by mygamingaccount** [comment]
This improved docking port design solves all issues with the ones discussed in this topic
![image](https://github.com/FakeFishGames/Barotrauma/assets/68995233/e910abac-bef9-43bc-8637-335f28b1941b)

**Comment by lIllIIlI** [comment]
Circuit boxes should have presets/loadouts you can save/load with a simple click. So even if someone griefs and deletes the contents, you only need to just reload the original setup.
Also, per user wiring permissions are really desired. I want my friends to be able to wire things and improve the sub, but not everyone who comes in with a screwdriver...

**Comment by Regalis11** [comment]
Addressed in https://github.com/FakeFishGames/Barotrauma-development/pull/5620 (PR in our private repo) by adding wiring permissions to the circuit boxes, closing
